nixos/modules/services/x11/window-managers/bspwm.nix

```diff
@@ -8,16 +8,36 @@ in
 
 {
   options = {
-    services.xserver.windowManager.bspwm.enable = mkEnableOption "bspwm";
+    services.xserver.windowManager.bspwm = {
+        enable = mkEnableOption "bspwm";
+        startThroughSession = mkOption {
+            type = with types; bool;
+            default = false;
+            description = "Start the window manager through the script defined in sessionScript. Defaults to bspwm-session";
+        };
+        sessionScript = mkOption {
+            default = null;
```

Comment:
you could pass the bspwm-session script as the default and remove one of the conditionals below